napari/_qt/_tests/test_qt_notifications.py

```diff
@@ -115,7 +119,9 @@ def run(self):
 
 @pytest.mark.parametrize('severity', NotificationSeverity.__members__)
 @patch('napari._qt.dialogs.qt_notification.QDialog.show')
-def test_notification_display(mock_show, severity, monkeypatch, qtbot):
+def test_notification_display(
+    mock_show, severity, monkeypatch, qtbot, clean_current
```

Comment:
The lack of these two `clean_current` fixtures may be the source of the problem.

Reply:
Maybe? The checks did all pass this time, but maybe we just got lucky?

If you think that did help and you understand the patching in `clean_current`, it would be great to add some explanatory comments there. I understand adding the notification dialog to `qtbot` (though I think it would just be simpler/more clear to do it in each test rather than patching `show` to do it), but don't understand the patch on main window and also would like to know if the commented patch on the animations should be uncommented or deleted.

Reply:
I introduce this. I do not understand now why it is required, but when removing it, the fail is reproduced on CI. 

I have added assert that should fail if there will be some hanging viewer. Lets investigate